rust/src/logger.rs

```diff
@@ -0,0 +1,176 @@
+// Licensed to the Software Freedom Conservancy (SFC) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The SFC licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use crate::metadata::now_unix_timestamp;
+use env_logger::fmt::Color;
+use env_logger::Target::Stdout;
+use log::Level;
+use log::LevelFilter::{Debug, Info, Trace};
+use serde::{Deserialize, Serialize};
+use std::cell::RefCell;
+use std::io::Write;
+use std::ops::Deref;
+use Color::{Blue, Cyan, Green, Red, Yellow};
+
+enum OutputType {
+    Logger,
+    Json,
+    Shell,
+}
+
+pub struct Logger {
```

Comment:
Is there no generic log utility in Rust? Just wondering if one needs to code the format. Minor thing from my side, though.

Reply:
Selenium Manager uses a crate called [env_logger](https://docs.rs/env_logger/latest/env_logger/), which is a popular Rust logger that can be configured via environment variables. The implementation of the custom `Logger` structure in Selenium Manager wraps that crate (for the `LOGGER` option selected in the CLI, which is the default one) and also, it gathers these logs to be flushed as the end as JSON format (for the `JSON` option selected in the CLI).

Reply:
👍